pipenv/project.py

```diff
@@ -631,6 +625,10 @@ def add_package_to_pipfile(self, package_name, dev=False):
             p[key] = {}
         package = convert_deps_from_pip(package_name)
         package_name = [k for k in package.keys()][0]
+        name = self.get_package_name_in_pipfile(package_name, dev)
+        if name and name != package_name:
+            # Replace the packge name
+            del p[key][name]
```

Comment:
If I'm understanding this part of the change correctly, I don't think it does what we want in the case that @jtratner mentions in https://github.com/pypa/pipenv/pull/1826#issuecomment-376758741, where `Pipfile` contains a stricter version specifier than `project_name="*"` and the install command is just `pipenv install project_name`.

The simplest fix I can see would be to just leave the denormalised name alone in the case where it's already present - the change to normalise keys on lookup in  `get_package_name_in_pipfile` will handle such cases.